src/core/net/srp_client.cpp

```diff
@@ -1428,6 +1460,75 @@ void Client::HandleTimer(void)
     }
 }
 
+#if OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE
+
+void Client::EnableAutoStartMode(AutoStartCallback aCallback, void *aContext)
+{
+    mAutoStartCallback = aCallback;
+    mAutoStartContext  = aContext;
+
+    VerifyOrExit(!mAutoStartModeEnabled);
```

Comment:
```suggestion
    VerifyOrExit(!mAutoStartModeEnabled);

    mAutoStartCallback = aCallback;
    mAutoStartContext  = aContext;
```

Reply:
The order here is intentional.
We want to allow user  to change the callback even when auto-strat is already enabled/running. 
Note that this method is `void` return so we allow it to be called anytime (no error returned if already running).